### PR TITLE
feat(mobile): Adding setting in mobile app to TLS client certificate

### PR DESCRIPTION
**Pull request by yjiang-c**
This is to add an advanced setting in Immich mobile app to import TLS client certificate and private key.

Some users deploy Immich server behind a reverse web proxy server like nginx. Through setting up reverse proxy server, immich client could be asked a TLS/SSL client certificate for authentication. This can add an additional secure layer to protect Immich server behind a reverse proxy.

### CODE REVIEW

**Comment by alextran1502** [comment]
Thank you for the request.

Can you help with screenshot for the UI change and a video capture of how to use the new feature?

File: mobile/lib/widgets/settings/ssl_client_cert_settings.dart [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+        ),
+      );
+      return;
+    }
+    SSLClientCertStoreVal.delete();
+    HttpOverrides.global = HttpSSLCertOverride();
+    showDialog(
+      context: context,
+      builder: (ctx) => AlertDialog(
+        content: const Text("Client certificate is removed!"),
```
Comment (alextran1502) [inline mobile/lib/widgets/settings/ssl_client_cert_settings.dart:100]: Can we have all the texts as  translation strings, please?

Comment (yjiang-c) [reply mobile/lib/widgets/settings/ssl_client_cert_settings.dart:100]: I just know English and Chinese. How do immich team generate translation? Through AI?

**Comment by yjiang-c** [comment]
> Thank you for the request.
> 
> Can you help with a screenshot for the UI change and a video capture of how to use the new feature?

The following four photos are for Import 
Screenshot_1720036163
Screenshot_1720036228
Screenshot_1720036246
Screenshot_1720036261

File: mobile/lib/widgets/settings/ssl_client_cert_settings.dart (hunk 1 above)
Comment (alextran1502) [reply mobile/lib/widgets/settings/ssl_client_cert_settings.dart:100]: You can just use English, we have other contributors help with the translation for other languages on a translation platform

Comment (yjiang-c) [reply mobile/lib/widgets/settings/ssl_client_cert_settings.dart:100]: Thanks! I will do it.

**Comment by yjiang-c** [comment]
> Thank you for the request.
> 
> Can you help with a screenshot for the UI change and a video capture of how to use the new feature?

**Comment by yjiang-c** [comment]
Sorry, I accidentally close this PR. I have created https://github.com/immich-app/immich/pull/10860 for the changes suggested in this PR. In addition, a demo video is enclosed in the new PR.

**Pull request closed without merging**